app/controllers/submissions_controller.rb

```diff
@@ -169,19 +147,60 @@ def update_plagiarism
         title: "",
         info: nil)
       sub_comment.save
-      # Copy any grades
-      @submission.grades.each do |g|
-        new_g = Grade.new(
-          grader_id: g.grader_id,
-          submission_id: sub.id,
-          grading_output: g.grading_output,
-          score: g.score,
-          out_of: g.out_of,
-          available: g.available)
-        new_g.save
-      end
     end
-    redirect_to course_assignment_path(@course, @assignment)
+    redirect_to course_assignment_path(@course, @assignment),
+                notice: "Submission marked as plagiarized for #{guilty_students.map{|u| User.find(u).name}.join(', ')}"
+  end
+
+  def split_submission
+    if @submission.users.count == 1
+      redirect_back course_assignment_submission_path(@course, @assignment, @submission),
+                    alert: "Submission is not a team-submission; no need to split it"
+      return
+    end
+    @submission.users.each do |student|
+      split_sub(@submission, student)
+    end
+    redirect_to course_assignment_path(@course, @assignment),
+                notice: "Group submission split for #{@submission.users.map(&:name).join(', ')}"
+  end
+
+  def split_sub(orig_sub, for_user, score = nil)
+    team = orig_sub.team
+    if team
```

Comment:
If there's no team, it still produces a self-consistent copy of the submission.  This helper can be called from update_plagiarism, too, which might be invoked on a submission with only a single student.

I don't see how anything could fail in this code sequence, but yes, it probably should be in a transaction.  I don't know how to write those in rails, though; I've not been able to figure it out from the documentation.  (I don't understand which class is the "right" one to invoke .transaction on, how nested transactions could be coded in practice, how to recover from them gracefully, or basically anything else about how to use that API idiomatically.)